server/aws-lsp-codewhisperer/src/language-server/agenticChat/agenticChatController.ts

```diff
@@ -2578,7 +2591,9 @@ export class AgenticChatController implements ChatHandlers {
 
         // Since model selection is mandatory, the only time modelId is not set is when the chat history is empty.
         // In that case, we use the default modelId.
-        let modelId = this.#chatHistoryDb.getModelId() ?? defaultModelId
+        // defaultModelId is undefined only when the clients does not enable model selection, even the
+        // chat history has model id(the case should not exist), we still need the modelId to be undefined
+        let modelId = defaultModelId === undefined ? undefined : (this.#chatHistoryDb.getModelId() ?? defaultModelId)
```

Comment:
Instead of storing this value in the agenticChatServer. We can retrieve whether modelSelection  is enabled by querying the values from `this.features.lsp.getClientInitializeParams()` and then passing it to the utility function (called out in a different comment). Then here we can set modelId to undefined when this is not enabled. This would remove the need to store any additional properties in the agenticServer and the reference for it in this class as well